lib/helper/WebDriver.js

```diff
@@ -640,6 +640,32 @@ class WebDriver extends Helper {
   async _locate(locator, smartWait = false) {
     if (require('../store').debugMode) smartWait = false;
 
+    // special locator type for Shadow DOM
+    if (locator.type === 'shadowDom' || locator.shadowDom) {
```

Comment:
Good, please refactor it to its own custom method `_locateShadowDom()`

Reply:
done